response_operations_ui/views/collection_exercise.py

```diff
@@ -1194,13 +1200,41 @@ def view_ci_versions(short_name: str, period: str, form_type: str) -> str:
 @login_required
 def save_ci_versions(short_name: str, period: str, form_type: str):
     ci_version = request.form.get("ci-versions")
+    ce_details = build_collection_exercise_details(short_name, period, include_ci=True)
```

Comment:
Again, a heavy operation to simply get the exercise_id. We will progress a separate strategy for this outside of this PR. We _could_ cache this `build_collection_exercise_details` method here, as long as we know to call it only for static data it contains, and not for anything that might change regularly. So cache it for the sole purpose of persisting the collection exercise ID.